src/widgets/resourceList.js

```diff
@@ -0,0 +1,519 @@
+import { Op, QueryTypes } from 'sequelize';
+import {
+  ActivityReport,
+  ActivityRecipient,
+  Grant,
+  OtherEntity,
+  Recipient,
+  sequelize,
+} from '../models';
+import { formatNumber } from './helpers';
+import { REPORT_STATUSES, RESOURCE_DOMAIN } from '../constants';
+
+export async function resourceData(scopes) {
+  // Query Database for all Resources within the scope.
+  const reports = await ActivityReport.findAll({
+    attributes: ['id', 'numberOfParticipants'],
+    where: {
+      [Op.and]: [
+        scopes.activityReport,
+        { calculatedStatus: REPORT_STATUSES.APPROVED },
+      ],
+    },
+    include: [
+      {
+        model: ActivityRecipient,
+        as: 'activityRecipients',
+        attributes: ['id'],
+        where: { [Op.and]: [scopes.activityRecipient] },
+        required: true,
+        include: [
+          {
+            model: Grant,
+            as: 'grant',
+            attributes: ['recipientId'],
+            where: { [Op.and]: [scopes.grant] },
+            required: true,
+            include: [
+              {
+                model: Recipient,
+                as: 'recipient',
+                attributes: ['id'],
+                where: { [Op.and]: [scopes.recipient] },
+                required: false,
+              },
+            ],
+          },
+          {
+            model: OtherEntity,
+            as: 'otherEntity',
+            attributes: ['id'],
+            where: { [Op.and]: [scopes.otherEntity] },
+            required: false,
+          },
+        ],
+      },
+    ],
+    raw: true,
+  });
+
+  const reportIds = reports.map((r) => r.id);
+  // to get correct escaping https://regex101.com/ code generator works well
+  const urlRegex = '(?:(?:http|ftp|https|file):\\/\\/)(?:www\\.)?(?:[\\w%_-]+(?:(?:\\.[\\w%_-]+)+)|(?:\\/[\\w][:]))(?:[\\w\\\\\'\'.,@?^=%&:\\/~+#()-]*[\\w@?^=%&\\/~+#-])';
+  const domainRegex = '^(?:(?:http|ftp|https|file):\\/\\/)?(?:www\\.)?((?:[\\w%_-]+(?:(?:\\.[\\w%_-]+)+)|(?:\\/[\\w][:])))';
+
+  const resources = await sequelize.query(`
+    WITH
+    "ARResources" AS (
+        SELECT
+            id "activityReportId",
+            "nonECLKCResourcesUsed",
+            "ECLKCResourcesUsed",
+            "createdAt",
+            "updatedAt"
+        FROM "ActivityReports" a
+        WHERE a.id in (${reportIds.join(',') || null})
```

Comment:
should we be running this query at all if there are no reportIds?

Reply:
and it does so several times, as it in the query in a few places.